src/ScreenCover/index.js

```diff
@@ -1,23 +1,61 @@
-import React, { Fragment } from 'react'
+import React, { Fragment, PureComponent } from 'react'
 import propTypes from 'prop-types'
 import css from 'styled-jsx/css'
 import cx from 'classnames'
 
-export const ScreenCover = ({ onClick }) => (
-    <div className="screen-cover" onClick={onClick}>
-        <style jsx>{`
-            div {
-                background: rgba(200, 200, 200, 0.6);
-                height: 100%;
-                left: 0;
-                position: absolute;
-                top: 0;
-                width: 100%;
-            }
-        `}</style>
-    </div>
-)
+class ScreenCover extends PureComponent {
+    elContainer = React.createRef()
+
+    componentDidMount() {
+        document.addEventListener('click', this.onDocClick)
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('click', this.onDocClick)
+    }
+
+    onDocClick = evt => {
+        console.log(this.elContainer, evt)
+        if (
+            this.elContainer.current &&
+            this.elContainer.current === evt.target
+        ) {
+            this.props.onClick()
+        }
+    }
+
+    render() {
+        return (
+            <div className="screen-cover" ref={this.elContainer}>
```

Comment:
Suppose we'd change this line to:
```html
<div className="screen-cover" onClick={this.props.onClick}>
```
Wouldn't that have the same effect as a click on the document? (considering the `div.screen-cover` will in fact cover the whole screen) And then this could be a stateless component 🎉 

But what if this `ScreenCover` has a loader in it and we want `props.onClick` to get triggered when clicking on that too? Then the loader should get `pointer-events: none;`. I think this would make sense.

Reply:
The `ScreenCover` is used for the `Modal` too, and using `pointer-events: none;` killed the interactions with the `Modal` itself (including the action buttons), so that felt a bit heavy handed.

Reply:
> killed the interactions with the Modal itself (including the action buttons)

Aha.... interesting. That could have to do with event propagation. A slightly different markup could fix things, something like this:
```jsx
<div className="screen-cover">
    <div className="backdrop"></div>
    { children && <div className="children">{this.props.children}</div>}
</div>
```
Because the backdrop is a separate div, the events in the children won't clash with the events in the backdrop.

Reply:
b0d8923